File: PagingRemoteMediatorJava/app/src/main/java/com/websarva/wings/android/pagingremotemediatorjava/ui/MainViewModel.java
package com.websarva.wings.android.pagingremotemediatorjava.ui;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModelKt;
import androidx.paging.Pager;
import androidx.paging.PagingData;
import androidx.paging.PagingLiveData;

import com.websarva.wings.android.pagingremotemediatorjava.data.local.Phone;
import com.websarva.wings.android.pagingremotemediatorjava.data.repository.PhoneRepository;

import kotlinx.coroutines.CoroutineScope;

public class MainViewModel extends AndroidViewModel {
	private PhoneRepository _phoneRepository;
	private LiveData<PagingData<Phone>> _phoneListLiveData;

	public MainViewModel(@NonNull Application application) {
		super(application);
		_phoneRepository = new PhoneRepository(application);

		Pager<Integer, Phone> phoneListPager = _phoneRepository.getAllPhoneListPager();
		CoroutineScope viewModelScope = ViewModelKt.getViewModelScope(MainViewModel.this);
		_phoneListLiveData = PagingLiveData.cachedIn(PagingLiveData.getLiveData(phoneListPager), viewModelScope);
	}

	public LiveData<PagingData<Phone>> getPhoneListLiveData() {
		return _phoneListLiveData;
	}
}

File: PagingRemoteMediatorKotlin/app/src/main/java/com/websarva/wings/android/pagingremotemediatorkotlin/ui/MainViewModel.kt
package com.websarva.wings.android.pagingremotemediatorkotlin.ui

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.viewModelScope
import androidx.paging.PagingData
import androidx.paging.cachedIn
import com.websarva.wings.android.pagingremotemediatorkotlin.data.local.Phone
import com.websarva.wings.android.pagingremotemediatorkotlin.data.repository.PhoneRepository
import kotlinx.coroutines.flow.Flow

class MainViewModel(application: Application) : AndroidViewModel(application) {
	val phoneListFlow: Flow<PagingData<Phone>>
	private val _phoneRepository: PhoneRepository

	init {
		_phoneRepository = PhoneRepository(application)
		val phoneListPager = _phoneRepository.getAllPhoneListPager()
		phoneListFlow = phoneListPager.flow.cachedIn(viewModelScope)
	}
}